src/components/Settings/Customization/Layout/LayoutSelector.tsx

```diff
@@ -68,97 +68,150 @@ export const LayoutSelector = ({ defaultLayout }: LayoutSelectorProps) => {
-          checked={leftSidebar}
-          onChange={(ev) => handleChange('enabledLeftSidebar', ev, setLeftSidebar)}
-        />
-        <Checkbox
-          label={t('layout.enablersidebar')}
-          description={t('layout.enablersidebardesc')}
-          checked={rightSidebar}
-          onChange={(ev) => handleChange('enabledRightSidebar', ev, setRightSidebar)}
-        />
-        <Checkbox
-          label={t('layout.enablesearchbar')}
-          checked={searchBar}
-          onChange={(ev) => handleChange('enabledSearchbar', ev, setSearchBar)}
-        />
-        <Checkbox
-          label={t('layout.enabledocker')}
-          checked={docker}
-          onChange={(ev) => handleChange('enabledDocker', ev, setDocker)}
-        />
-        <Checkbox
-          label={t('layout.enableping')}
-          checked={ping}
-          onChange={(ev) => handleChange('enabledPing', ev, setPing)}
-        />
+          {rightSidebar && (
+            <Paper className={classes.secondaryWrapper} p="xs" withBorder>
+              <Flex gap={5} align="start" wrap="wrap">
+                {createDummyArray(5).map((item, index) => (
+                  <PlaceholderElement
+                    height={30}
+                    width={index % 4 === 0 ? 60 + 5 : 30}
+                    key={`example-item-right-sidebard-${index}`}
+                    index={index}
+                    hasPing={enabledPing}
+                  />
+                ))}
+              </Flex>
+            </Paper>
+          )}
+        </Flex>
+
+        <Divider label="Layout options" labelPosition="center" mt="md" mb="xs" />
+        <Stack spacing="xs">
+          <Checkbox
+            label={t('layout.enablelsidebar')}
+            description={t('layout.enablelsidebardesc')}
+            checked={leftSidebar}
+            onChange={(ev) => handleChange('enabledLeftSidebar', ev, setLeftSidebar)}
+          />
+          <Checkbox
+            label={t('layout.enablersidebar')}
+            description={t('layout.enablersidebardesc')}
+            checked={rightSidebar}
+            onChange={(ev) => handleChange('enabledRightSidebar', ev, setRightSidebar)}
+          />
+          <Checkbox
+            label={t('layout.enablesearchbar')}
+            checked={searchBar}
+            onChange={(ev) => handleChange('enabledSearchbar', ev, setSearchBar)}
+          />
+          <Checkbox
+            label={t('layout.enabledocker')}
+            checked={docker}
+            onChange={(ev) => handleChange('enabledDocker', ev, setDocker)}
+          />
+          <Checkbox
+            label={t('layout.enableping')}
+            checked={ping}
+            onChange={(ev) => handleChange('enabledPing', ev, setPing)}
+          />
+        </Stack>
       </Stack>
-    </Stack>
+    </>
+  );
+};
+
+const PlaceholderElement = (props: any) => {
+  const { colorScheme, colors } = useMantineTheme();
+  const { height, width, hasPing, index } = props;
+
+  const Element = () => (
```

Comment:
Don't create a component inside a component! This component will be created on each rerun of the placeholder element.